core/txpool/locals/pool_journaler.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2025 The op-geth Authors
+// This file is part of the op-geth library.
+//
+// The op-geth library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The op-geth library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the op-geth library. If not, see <http://www.gnu.org/licenses/>.
+
+package locals
+
+import (
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/core/txpool"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// A PoolJournaler periodically journales a transaction pool to disk.
+// It is meant to be used instead of the TxTracker when the noLocals and
+// journalRemotes settings are enabled.
+//
+// OP-Stack addition.
+type PoolJournaler struct {
```

Comment:
I chose to add an alternative service to the new `TxTracker` instead of hacking the journalremotes feature into it, because that seemed more messy and less modular and maintainable to me. This way, our journalremotes feature is nicely encapsulated in a separate service.